src/alchemlyb/tests/parsing/test_namd.py

```diff
@@ -36,13 +48,183 @@ def test_u_nk_idws():
+
+    yield dataset
+    for fname in temp_fnames:
+        unlink(fname)
+
+
+@pytest.fixture
+def restarted_dataset_toomany_lambda2():
+    """Returns intentionally messed up dataset, where there are too many lambda2 values for a
+    given lambda1."""
+    dataset = load_restarted()
+    filenames = dataset['data']['both']
+
+    # For the same l1 and lidws we retain old lambda2 values thus ensuring a collision
+    def func_new_line(l):
+        l[LAMBDA1_IDX_NEW] = '0.2'
+        if len(l) > 9 and l[9] == 'LAMBDA_IDWS':
+            l[LAMBDA_IDWS_IDX_NEW] = '0.1'
+        return l
+
+    def func_free_line(l):
+        l[LAMBDA1_IDX_FREE] = '0.2'
+        return l
+
+    temp_fnames = []
+    for i in range(len(filenames)):
+        fname = _corrupt_fepout(filenames[i], [('#NEW', func_new_line), ('#Free', func_free_line)])
+        dataset['data']['both'][i] = fname
+        temp_fnames.append(fname)
+
+    yield dataset
+    for fname in temp_fnames:
+        unlink(fname)
+
+
+@pytest.fixture
+def restarted_dataset_toomany_lambda_idws():
+    """Returns intentionally messed up dataset, where there are too many lambda2 values for a
+    given lambda1."""
+    dataset = load_restarted()
+    filenames = dataset['data']['both']
+
+    # For the same lambda1 and lambda2 we retain old lambda_idws values thus ensuring a collision
+    changed = False
+    def func_new_line(l):
+        # Ensure that changing these lamda values won't cause a reversal in direction and trigger
+        # an exception we're not trying to test here
+        if len(l) > 9 and float(l[LAMBDA_IDWS_IDX_NEW]) < 0.5:
+            l[LAMBDA1_IDX_NEW], l[LAMBDA2_IDX_NEW] = '0.5', '0.6'
+            changed = True
+        return l
+
+    def func_free_line(l):
+        if changed:
+            l[LAMBDA1_IDX_FREE], l[LAMBDA2_IDX_FREE] = '0.5', '0.6'
+        return l
+
+    temp_fnames = []
+    for i in range(len(filenames)):
+        fname = _corrupt_fepout(filenames[i], [('#NEW', func_new_line)])
+        dataset['data']['both'][i] = fname
+        temp_fnames.append(fname)
+
+    yield dataset
+    for fname in temp_fnames:
+        unlink(fname)
+
+
+@pytest.fixture
+def restarted_dataset_direction_changed():
+    """Returns intentionally messed up dataset, with one window where the lambda values are reversed."""
+    dataset = load_restarted()
+    filenames = dataset['data']['both']
+
+    def func_new_line(l):
+        l[6], l[8], l[10] = l[10], l[8], l[6]
+        return l
+
+    def func_free_line(l):
+        l[7], l[8] = l[8], l[7]
+        return l
+    
+    # Reverse the direction of lambdas for this window
+    idx_to_corrupt = filenames.index(sorted(filenames)[-3])
+    fname1 = _corrupt_fepout(filenames[idx_to_corrupt], [('#NEW', func_new_line), ('#Free', func_free_line)])
+    dataset['data']['both'][idx_to_corrupt] = fname1
+    yield dataset
+
+    unlink(fname1)
+
+
+def test_u_nk_restarted(restarted_dataset, restarted_dataset_direction_changed,
```

Comment:
Make this one test for each of these different exceptions: one dataset, one exception test